usaspending_api/common/threaded_data_loader.py

```diff
@@ -0,0 +1,194 @@
+from django.db.models import Q
+from django.conf import settings
+from django.db import connection
+from retrying import retry
+import time
+import csv
+import queue
+import threading
+
+
+# This class is a threaded data loader
+class ThreadedDataLoader():
+    # The threaded data loader requires a bit of set up, and explanation of the
+    # parameters are below. Most parameter defaults are about where you'd want them:
+    #   model_class - The class of the model each row of the file corresponds to
+    #   threads - The number of threads to use. Default: 10
+    #   field_map - A dict map from the CSV's columns to model field names. Default: Empty
+    #   value_map - A dict map of default or processed values to use. Keys should be model fields
+    #               To process values, create a lambda function that accepts a parameter which is
+    #               a dict representing the row. i.e. lambda row: datetime(row['published date'])
+    #               (Theoretically you can use some arbitrary method, as well, for complex processes)
+    #   collision_field - The field with which to collide upon, usually some PK or unique identifier. String
+    #                     Default: None
+    #   collision_behavior - A string representing the collision behavior, which occurs when a collision is detected
+    #                       Options:
+    #                           delete - Delete the row from the data store and load new row
+    #                           update - Update the row in the data store with new values from the CSV
+    #                           skip - Skip the row
+    #                           skip_and_complain - Log a warning and skip the row
+    #                           die - Raise an exception and cease execution
+    #   post_row_function - A function to call after every row. Should support parameters of row and instance
+    #                       i.e. def do_this_after_a_row(row=None, instance=None)
+    #   pre_row_function - Like post_row_function, but before the model class is updated
+    #   post_process_function - A function to call when all rows have been processed, uses the same
+    #                           function parameters as post_row_function
+    def __init__(self, model_class, threads=10, field_map={}, value_map={}, collision_field=None, collision_behavior='delete', pre_row_function=None, post_row_function=None, post_process_function=None):
```

Comment:
it seems like `update` should be the default for `collision_behavior`. Any particular reason it's `delete`?